byway/src/main/webapp/recommendations.js

```diff
@@ -224,4 +243,13 @@ function placeMarker(place) {
   marker.addListener("click", () => {
     infoWindow.open(map, marker);
   });
+  markers.push(marker);
+}
+
+/* Iterate through all markers and set their maps to null. Make Markers array empty. */
+function clearMarkers() {
+  for (let i = 0; i < markers.length; i++) {
+    markers[i].setMap(null);
+  }
+  markers = [];
```

Comment:
You are iterating through every marker and resetting the map, but then you are reassigning the value of markers? Why do you need to reset the map of the marker objects if you are just resetting the array anyways? 

If the objects inside the map are going to be discarded, then it doesn't make sense to go through and edit them first. Unless a copy of the object are held in some other data structure, in which case, there are ways you could make this more clear in the code. Without any context, this code looks like it doing something pointless.

Reply:
Setting the map to null removes the marker object from the google.maps object, but it still lived in the markers array. I still had to set it to an empty array to completely delete it (https://developers.google.com/maps/documentation/javascript/markers#remove)

I updated this function to either show or hide markers based on a boolean that is passed in. We will use this to add a toggle feature to either show markers or hide them on click events.